papers/yellow/main.tex

```diff
@@ -748,19 +727,71 @@ \subsection{Signing A Transaction}
 
 
 
+\subsection{Transaction Validation Rules}
+\label{tx-validation}
 
-\subsection{Verifying A Transaction}
+We split validation of a single transaction into two stages. There are $stateless checks$ which could be done by the
+transaction only being presented. Stateful checks are requiring knowledge of the current state, in the form of the whole
+UTXO set or a part of it, namely concrete boxes a transaction is destroying along with proof of authenticity for them~(
+against a root hash included into a last block header).
 
+Stateless checks are:
 
+\begin{itemize}
+    \item{} a transaction must spend at a least one input and create at least one output. A transaction spends no more
+    than $32767$ inputs and creates no more than $32767$ outputs.
+    \item{} all the output amounts must be non-negative.
+    \item{} an output can not contain more than 4 assets. All the assets amounts must be positive.
+    \item{} transaction outputs collectively could not contain more than 16 assets.
+\end{itemize}
+
+\knote{Check box and transaction sizes.}
+
+Stateful checks are:
+
+\begin{itemize}
+    \item{} total input and output ergo amounts must be the same. Adding up should be done with overflow checks.

```

Comment:
I think first check should looks like `All inputs refers to existing boxes in UTXO`.